### ISSUE TITLE
How to find label from the estimator predict call for ranker eval purposes

### ISSUE THREAD

**Issue by vitalyli** [tensorflow/ranking#284]
eval_input_fn feed a tuple of features, label. 
Could someone share how to find corresponding label from the estimator predict call for the ranker eval purposes..?

    for prediction in estimator.predict(eval_input_fn):
        doc_count = len(prediction)

        print(prediction)

        smax = stable_softmax(prediction)

        rank_idx = np.argmax(smax)

**Comment by ramakumar1729** [comment]
@vitalyli the predict call function returns the scores for each example in ELWC. It does not take the label as input nor returns it. If you want to access the label, you can access it directly from the `eval_input_fn`.

**Comment by vitalyli** [comment]
eval_input_fn is a function pointer right, the estimator base class take it and starts pulling stream of samples from my test set. Each results in that stream would have a tuple: (features,label). The issue with predict call, the estimator is tossing out label and returns predictions only. Which I understand why it's hiding it to keep everyone honest. However I want to be able to do 
softmax then argmax and get index of predicted ranking idex. And then compare that with the label index. 
Estimator currently doesn't allow to override this behavior in any way. 
Meanwhile the order in which samples come out from the eval_input_fn is via a dataset that pulls it from N files it finds on disk. The oder of what estimator is processing is not clear if that can be repeated by running second loop outside of estimator in order to align label with predictions. My ask is if predict was could have an option to return label indexes as was supplied by eval_input_fn. The default and only mode currently is it's tossing labels without way to get to them.
